### DISCUSSION TITLE
Roadmap poll

### DISCUSSION THREAD

**Discussion by ansibleguy** [O-X-L/ansible-opnsense#14] [Polls]
Greetings!

It's about time the project got new goals.

I would like to ask you what functionality would be useful to you.

Only [API-enabled services](https://docs.opnsense.org/development/api.html) can be implemented!

If any unlisted functionalities would be of interest to you => comment (:

----

**References**:

- [Captive Portal](https://docs.opnsense.org/development/api/core/captiveportal.html)
- [FreeRadius](https://docs.opnsense.org/development/api/plugins/freeradius.html)
- [Diagnostics](https://docs.opnsense.org/development/api/core/diagnostics.html)
- [IDS](https://docs.opnsense.org/development/api/core/ids.html)
- [OpenVPN](https://docs.opnsense.org/development/api/core/openvpn.html)
- [Zabbix Agent](https://docs.opnsense.org/development/api/plugins/zabbixagent.html)/[Zabbix Proxy](https://docs.opnsense.org/development/api/plugins/zabbixproxy.html)
- [Tor](https://docs.opnsense.org/development/api/plugins/tor.html)
- [ClamAV](https://docs.opnsense.org/development/api/plugins/clamav.html)

**Comment by ansibleguy** [comment]
Open until mid of may.

**Reply by mircea-pavel-anton** [reply]
I would also be interested in the node_exporter package.

As for the DHCP server, I don't see anything about it being API configurable, so I assume we can't set DHCP ranges and static leases?

**Reply by ansibleguy** [reply]
Hi.

The node-exporter is a good idea - yes.

There is no API for managing DHCP-config or settings yet. :(

**Comment by ansibleguy** [comment]
NOTE: The OpenVPN implementation is very limited by the API!
Only 'Client profile export' and 'Service actions' (start/stop/restart/status/killSession/searchSession) can be performed.

\- AnsibleGuy

**Reply by yurtesen** [reply]
Some new OpenVPN API functionality have been added.
https://docs.opnsense.org/development/api/core/openvpn.html

**Reply by ansibleguy** [reply]
Yeah - have seen the changes already.
But haven't found time to implement any :/

**Comment by ansibleguy** [comment]
Alright - next ones will be:
- IPS/IDS
- OpenVPN

\- AnsibleGuy
